### PR TITLE
Handle missed BME case in resolveInvokeDynamic for OJDK MHs

### PR DESCRIPTION
**Pull request by ThanHenderson**
This patch fixes a missed `BootstrapMethodError`-wrapping case for OJDK MHs in `MethodHandleNatives.resolveInvokeDynamic` when there is an error thrown during `MethodType` resolution.
    
Signed-off-by: Nathan Henderson <nathan.henderson@ibm.com>

### CODE REVIEW

**Comment by ThanHenderson** [comment]
Here are the builds: https://hyc-runtimes-jenkins.swg-devops.com/job/Pipeline-Build-Test-Personal/19729/

Failure seen in JDK8:

- https://github.com/eclipse-openj9/openj9/issues/15862 

Failure seen in JDK11 are network timeouts.


**tldr** OJDK MHs for 8 and 11 also pass `system` and `jck` suites

**Comment by babsingh** [comment]
jenkins test sanity.functional,sanity.openjdk plinux jdk8,jdk17

File: jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java [hunk 1]
```diff
@@ -297,6 +297,11 @@ private static final Object resolveInvokeDynamic(long j9class, String name, Stri
...
 			}
 /*[ENDIF] JAVA_SPEC_VERSION < 11*/
+
+			if (type == null) {
+				throw new BootstrapMethodError(e);
+			}
+
 			/* Any throwables are wrapped in an invoke-time BootstrapMethodError exception throw. */
 			try {
 				MethodHandle resultHandle;
```
Comment (babsingh) [inline jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java:307]: Unused variable can be removed.

**Comment by babsingh** [comment]
jenkins test sanity.functional,sanity.openjdk plinux jdk8,jdk11,jdk17

**Comment by babsingh** [comment]
jenkins test sanity.functional,sanity.openjdk zlinuxojdk292 jdk8,jdk11

**Comment by ThanHenderson** [comment]
The build failures are caused by adding the suggested changes here: https://github.com/eclipse-openj9/openj9/pull/18474/files#r1402752007

I will partially revert the changes to retain the `resultHandle` local to avoid odd inline casting.

**Comment by babsingh** [comment]
jenkins test sanity.functional,sanity.openjdk zlinuxojdk292 jdk8,jdk11

**Comment by ThanHenderson** [comment]
Test failures:

- https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.openjdk_s390x_linux_ojdk292_Personal_testList_1/5/consoleFull
- RPC failure for ojdk8 `sanity.functional` s390x
- JitServer timeout for ojdk17 `sanity.functional` ppc64

**Comment by babsingh** [comment]
LGTM. PR build failures are known and unrelated to this PR:
- https://github.com/eclipse-openj9/openj9/issues/14594
- https://github.com/eclipse-openj9/openj9/issues/17323
- Infra issue: `git fetch` failed

**Pull request merged**